src/_pytest/config/findpaths.py

```diff
@@ -86,14 +86,14 @@ def get_common_ancestor(paths: Iterable[py.path.local]) -> py.path.local:
     return common_ancestor
 
 
-def get_dirs_from_args(args):
-    def is_option(x):
-        return str(x).startswith("-")
+def get_dirs_from_args(args: List[str]) -> List[py.path.local]:
```

Comment:
```suggestion
def get_dirs_from_args(args: Iterable[str]) -> List[py.path.local]:
```

Reply:
That makes Str a valid argument as far as typing is concerned

Reply:
Right, an unfortunate Python legacy...

I think this is more a thing for Python/mypy to fix -- `Sequence` has the same issue, and `List` as input is not great as it implies that it will be modified.

I can change it back though, let me know.